hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/SqlJoinTest.java

```diff
@@ -0,0 +1,84 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql;
+
+import com.hazelcast.jet.sql.impl.connector.test.TestBatchSqlConnector;
+import com.hazelcast.sql.SqlService;
+import com.hazelcast.sql.impl.QueryException;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class SqlJoinTest extends SqlTestSupport {
+
+    private static SqlService sqlService;
+
+    @BeforeClass
+    public static void setUpClass() {
+        initialize(2, null);
+        sqlService = instance().getSql();
+    }
+
+    @Test
+    public void test_leftJoinFails() {
+        TestBatchSqlConnector.create(sqlService, "b", 0);
+
+        assertThatThrownBy(() -> sqlService.execute("SELECT 1 FROM b AS b1 LEFT JOIN b AS b2 ON b1.v = b2.v"))
+                .hasCauseInstanceOf(QueryException.class)
+                .hasMessageContaining("LEFT join not supported");
+
+        assertThatThrownBy(() -> sqlService.execute("SELECT 1 FROM b AS b1 LEFT OUTER JOIN b AS b2 ON b1.v = b2.v"))
+                .hasCauseInstanceOf(QueryException.class)
+                .hasMessageContaining("LEFT join not supported");
+    }
+
+    @Test
+    public void test_rightJoinFails() {
+        TestBatchSqlConnector.create(sqlService, "b", 0);
+
+        assertThatThrownBy(() -> sqlService.execute("SELECT 1 FROM b AS b1 RIGHT JOIN b AS b2 ON b1.v = b2.v"))
+                .hasCauseInstanceOf(QueryException.class)
+                .hasMessageContaining("RIGHT join not supported");
+
+        assertThatThrownBy(() -> sqlService.execute("SELECT 1 FROM b AS b1 RIGHT OUTER JOIN b AS b2 ON b1.v = b2.v"))
+                .hasCauseInstanceOf(QueryException.class)
+                .hasMessageContaining("RIGHT join not supported");
+    }
+
+    @Test
+    public void test_fullJoinFails() {
+        TestBatchSqlConnector.create(sqlService, "b", 0);
+
+        assertThatThrownBy(() -> sqlService.execute("SELECT 1 FROM b AS b1 FULL JOIN b AS b2 ON b1.v = b2.v"))
+                .hasCauseInstanceOf(QueryException.class)
+                .hasMessageContaining("FULL join not supported");
+
+        assertThatThrownBy(() -> sqlService.execute("SELECT 1 FROM b AS b1 FULL OUTER JOIN b AS b2 ON b1.v = b2.v"))
+                .hasCauseInstanceOf(QueryException.class)
+                .hasMessageContaining("FULL join not supported");
+    }
+
+    @Test
+    public void test_crossJoinFails() {
+        TestBatchSqlConnector.create(sqlService, "b", 0);
+
+        assertThatThrownBy(() -> sqlService.execute("SELECT 1 FROM b AS b1 CROSS JOIN b AS b2"))
```

Comment:
This query can be transformed to 
```sql
SELECT 1 FROM b AS b1 JOIN b AS b2 ON 1=1
```
or to:
```sql
SELECT 1 FROM b AS b1, b AS b2
```
I'm surprised that cross join isn't handled by Calcite before we observe it...

Reply:
Enabled `CROSS JOIN`s